app/assets/javascripts/mumuki_laboratory/application/profile.js

```diff
@@ -26,11 +26,17 @@ mumuki.load(function() {
   });
 
   function toggleEditButtonIfThereAreChanges() {
-    if (dataChanged() || avatarChanged()) {
-      $editButton.prop('disabled', false);
-    } else {
-      $editButton.prop('disabled', true);
-    }
+    let shouldEnable = requiredFieldsAreFilled() && (dataChanged() || avatarChanged());
+
+    $editButton.prop('disabled', !shouldEnable);
+  }
+
+  function requiredFieldsAreFilled() {
+    return Object.values($userForm.find('select, textarea, input').map(function () {
+      if ($(this).prop('required')) {
+        return !!$(this).val();
```

Comment:
When using an `$element` more than once, please reuse it using a local variable. Regardless of aesthetics, this avoids creation and evaluation of unnecessary objects